### ISSUE TITLE
How to quickly determine database connectivity? 🤷

### ISSUE THREAD

**Issue by Kukulkano** [jackc/pgx#981]
Hi, I'm using pgx to access a CockroachDB as backend. I utilize the pool function. 

I wonder how to quickly find out if the connected database node is still fine? Do I need to make a select to known data? Or a simple select for time? But this will very likely hit a timeout then and blocks for a while if the node is down. What is the recommended way to quickly proove database connection health without having timeouts?

**Comment by gomezjdaniel** [comment]
Use `Ping()`

**Comment by Kukulkano** [comment]
Thanks, I did not see before. Now the only global object I have is my **ConnPool** object. I can not call Ping() from there. Is there a way to get the underlying context from **ConnPool** object so that I can use Ping() command on that?

**Comment by gomezjdaniel** [comment]
If you don't have any context in the scope where you intent to call Ping `context.Background()` should be ok!

**Comment by Kukulkano** [comment]
Sorry, I'm new to golang and pgx. I tried like this, but I only get a blocking call that waits for minutes.

```
var p pgx.Conn
err := p.Ping(context.Background())
if err != nil {
```

The only available object is my connection pool of type ConnPool, which does not offer a Ping() function.

I found myself now trying like this:

```
_, err := MyPool.Exec(";")
if err != nil {
	return c.String(http.StatusServiceUnavailable, "Service Unavailable")
}
```

It looks like there is no timeout if I stop the CockroachDB service. It immediately fails and this looks good to me. Is this a good way to validate database availability?
